src/main/java/org/jenkinsci/plugins/mesos/MesosComputerLauncher.java

```diff
@@ -107,12 +107,27 @@ public void failed(JenkinsSlave slave) {
       // Since we just launched a slave, remove it from pending deletion in case it was marked previously while
       // we were waiting for resources to be available
       computer.getNode().setPendingDelete(false);
+      waitForSlaveConnection(computer);
       logger.println("Successfully launched slave" + name);
     }
 
     LOGGER.info("Finished launching slave computer " + name);
   }
 
+  private void waitForSlaveConnection(MesosComputer computer) {
+    while (computer.isOffline() && computer.isConnecting()) {
+      try {
+        LOGGER.info("Waiting for slave computer connection " + name);
```

Comment:
I think 'FINE' would be enough otherwise it will be too verbose for regular use.